src/hooks/useMulticallProvider.ts

```diff
@@ -0,0 +1,15 @@
+import { UseQueryResult, useQuery } from "@tanstack/react-query"
+
+import { MulticallProvider } from "../types/ethcall"
+import { getMulticallProvider } from "../utils"
+import { useActiveWeb3React } from "../hooks"
+
+export const useMulticallProvider = (): UseQueryResult<MulticallProvider> => {
+  const { chainId, library } = useActiveWeb3React()
+
+  return useQuery(["multicallProvider"], async () => {
+    if (!library || !chainId) throw new Error("library or chainId not found")
```

Comment:
UseQuery just let's you throw and then handles it? I love that

Reply:
Yes, when within the useQuery asynchronous fn parameter. It will then manage the error.